scarlet/source.py

```diff
@@ -119,24 +121,28 @@ def __init__(
             Multiple of the backround RMS used as a
             flux cutoff for morphology initialization.
         """
+        if spectrum is None:
+            spectrum = get_pixel_sed(sky_coord, observations)
 
         # initialize from observation
-        sed, morph, bbox, thresh = init_starlet_source(
+        morph, thresh_starlet = init_starlet_source(
             sky_coord,
             model_frame,
             observations,
-            coadd,
-            bg_cutoff,
-            thresh=thresh,
-            symmetric=True,
+            spectrum,
+            coadd=coadd,
+            coadd_rms=bg_cutoff,
+            symmetric=False,
             monotonic="angle",
             min_grad=min_grad,
             starlet_thresh=starlet_thresh,
         )
-        spectrum = TabulatedSpectrum(model_frame, sed, bbox=bbox[0])
+
+        spectrum = TabulatedSpectrum(model_frame, spectrum, step = 1.e-6)
```

Comment:
This is where you should create a special `Parameter` instance with the small step size.

Reply:
Addressed